164829200 descriptive validation error messages for registration

vino-jasuba opened the pull request:
#### Description
Currently, when a user attempts registration and has submitted invalid data, they get generic responses that do not help them quickly determine what they need to change in order for the registration request to go through.

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [ ] End to end
- [ ] Integration

#### Checklist:
- [ ] My code follows the style guide for this project
- [ ] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
164829200

ValerieRono committed:
164829199(jwt): implement jwt authentication

- ensure user model has a generate jwt token method
- ensure user should receive a JWT as the response upon successful log in or sign up

[Delivers 164829199]

vino-jasuba committed:
164829200-feature-(validation): add descriptive error messages

- ensure registration errors have descriptive error messages
- add validation tests

[Finishes 164829200]

ogiste commented on authors/apps/authentication/serializers.py:
```diff
@@ -1,10 +1,22 @@
+    :return: given string or raises a serializers.ValidationError
+    """
+
+    return RegexValidator(
```
We would need to change the Regex validator for the password, as it currently is it only allows alphanumeric characters. The password should have **at least** both alphabetic and numeric characters to be a valid password but should also allow for special characters which make a stronger password

vino-jasuba committed:
164829200-feature-(validation): update password validation rule

- ensure valid passwords have at least 1 lowercase, 1 uppercase and one special character

[Finishes 164829200]

ValerieRono reviewed (commented):
Add tests to improve coverage

vino-jasuba commented:
> Add tests to improve coverage

The changes that introduced the drop in coverage were not made by me. But yeah, I'll add the tests 👍
